### PR TITLE
[Merged by Bors] - feat: port Algebra.Order.Monoid.Basic

### PR DESCRIPTION
**Pull request by javra**
mathlib3 SHA: 9b2660e1b25419042c8da10bf411aa3c67f14383

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
In the future, please put the original autoported file in a separate commit, so people can easily see the manual changes

File: Mathlib/Algebra/Order/Monoid/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    (hsup : ∀ x y, f (x ⊔ y) = max (f x) (f y)) (hinf : ∀ x y, f (x ⊓ y) = min (f x) (f y)) :
+    LinearOrderedCommMonoid β :=
+  { hf.orderedCommMonoid f one mul npow, LinearOrder.lift f hf hsup hinf with }
+#align function.injective.linear_ordered_comm_monoid Function.Injective.linearOrderedCommMonoid
+
+-- TODO find a better home for the next two constructions.
+/-- The order embedding sending `b` to `a * b`, for some fixed `a`.
+See also `order_iso.mul_left` when working in an ordered group. -/
+@[to_additive
+      "The order embedding sending `b` to `a + b`, for some fixed `a`.\n
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/Basic.lean:56]:
```suggestion
      "The order embedding sending `b` to `a + b`, for some fixed `a`.
```

File: Mathlib/Algebra/Order/Monoid/Basic.lean [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  simps]
+def OrderEmbedding.mulLeft {α : Type _} [Mul α] [LinearOrder α]
+    [CovariantClass α α (· * ·) (· < ·)] (m : α) : α ↪o α :=
+  OrderEmbedding.ofStrictMono (fun n => m * n) fun _ _ w => mul_lt_mul_left' w m
+#align order_embedding.mul_left OrderEmbedding.mulLeft
+
+/-- The order embedding sending `b` to `b * a`, for some fixed `a`.
+See also `order_iso.mul_right` when working in an ordered group. -/
+@[to_additive
+      "The order embedding sending `b` to `b + a`, for some fixed `a`.\n
```
Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Order/Monoid/Basic.lean:67]:
```suggestion
      "The order embedding sending `b` to `b + a`, for some fixed `a`.
```

**Review by Ruben-VandeVelde — APPROVED** [review]

**Comment by ChrisHughes24** [comment]
bors merge

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**